masonry/src/widget/sized_box.rs

```diff
@@ -512,5 +554,131 @@ mod tests {
+
+        assert_debug_snapshot!(harness.root_widget());
+        assert_render_snapshot!(harness, "label_box_with_outer_padding");
+    }
+
     // TODO - add screenshot tests for different brush types
 }
+
+// --- MARK: PADDING STRUCT ---
+
+/// Padding specifies the spacing between the edges of the box and the child view.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Padding {
+    /// The amount of padding for the top edge.
+    pub top: f64,
+    /// The amount of padding for the trailing edge.
+    ///
+    /// For LTR contexts this is the right edge, for RTL it is the left edge.
+    pub trailing: f64,
+    /// The amount of padding for the bottom edge.
+    pub bottom: f64,
+    /// The amount of padding for the leading edge.
+    ///
+    /// For LTR contexts this is the left edge, for RTL it is the right edge.
+    pub leading: f64,
+}
+
+impl Padding {
+    /// Creates a new [Padding] object by specifying the amount of padding for each edge.
+    pub const fn new(top: f64, trailing: f64, bottom: f64, leading: f64) -> Self {
+        Self {
+            top,
+            trailing,
+            bottom,
+            leading,
+        }
+    }
+
+    /// A padding of zero for all edges.
+    pub const ZERO: Padding = Padding::all(0.);
+
+    /// Creates a new [Padding] object with equal amount of padding for all edges.
+    pub const fn all(padding: f64) -> Self {
+        Self::new(padding, padding, padding, padding)
+    }
+
+    /// Creates a new [Padding] object with the same amount of padding for the horizontal edges,
+    /// and zero padding for the vertical edges.
+    pub const fn horizontal(padding: f64) -> Self {
+        Self::new(0., padding, 0., padding)
+    }
+
+    /// Creates a new [Padding] object with the same amount of padding for the vertical edges,
+    /// and zero padding for the horizontal edges.
+    pub const fn vertical(padding: f64) -> Self {
+        Self::new(padding, 0., padding, 0.)
+    }
+
+    /// Creates a new [Padding] object with padding only at the top edge and zero padding for all other edges.
+    pub const fn top(padding: f64) -> Self {
+        Self::new(padding, 0., 0., 0.)
+    }
+
+    /// Creates a new [Padding] object with padding only at the trailing edge and zero padding for all other edges.
+    pub const fn trailing(padding: f64) -> Self {
+        Self::new(0., padding, 0., 0.)
+    }
+
+    /// Creates a new [Padding] object with padding only at the bottom edge and zero padding for all other edges.
+    pub const fn bottom(padding: f64) -> Self {
+        Self::new(0., 0., padding, 0.)
+    }
+
+    /// Creates a new [Padding] object with padding only at the leading edge and zero padding for all other edges.
+    pub const fn leading(padding: f64) -> Self {
+        Self::new(0., 0., 0., padding)
+    }
+}
+
+impl From<f64> for Padding {
+    /// Converts the value to a [Padding] object with that amount of padding on all edges.
```

Comment:
I've looked at the generated docs page, and these are really hard to discover.

These docs are in the right place, but we should also mention these impls at the top level of `Padding`

Reply:
I'm unsure how to best do this?

Reply:
As in, adding some text to the doc comment on `Padding` describing that these implementations exist for convenience?

Reply:
Okay, I added this comment now, how does that look?

https://github.com/linebender/xilem/pull/736/files#diff-5cd89e6ede9989e07db2161ca9bb872fc743fe9ecfe65f1e43da72fa6379079cR30-R31

Reply:
I'd probably choose slightly different wording, but that form looks good. Thanks!